### PR TITLE
[TorchToTosa] Lower zero-K matmul to zero const

### PR DESCRIPTION
**Pull request by catcor01**
## Summary
- allow Torch-to-TOSA matmul lowering to handle static zero contraction when the result shape is non-empty
- lower those cases to a typed zero `tosa.const` instead of rejecting the zero-sized input
- keep zero-sized output tensors rejected as unsupported by TOSA

### CODE REVIEW

**Review by Lallapallooza — CHANGES_REQUESTED** [review]
Please add e2e coverage.

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/matmul.py [hunk 1]
```diff
@@ -342,6 +342,27 @@ def AtenMmFloatTypes_basic(module, tu: TestUtils):
 # ==============================================================================
 
 
+class AtenMmZeroK(torch.nn.Module):
```
Comment (Lallapallooza) [inline projects/pt1/python/torch_mlir_e2e_test/test_suite/matmul.py:369]: Can we add `aten.matmul` and `aten.bmm` coverage as well.

File: test/Conversion/TorchToTosa/basic.mlir [hunk 2]
```diff
@@ -4877,6 +4877,41 @@ func.func @torch.aten.mm$bf16(%arg0: !torch.vtensor<[1,22],bf16>, %arg1: !torch.
...
+
+// -----
+// CHECK-LABEL:   func.func @torch.aten.matmul$zero_k_f32(
+// CHECK-SAME:      %[[LHS:.*]]: !torch.vtensor<[5,0],f32>,
+// CHECK-SAME:      %[[RHS:.*]]: !torch.vtensor<[0,10],f32>) -> !torch.vtensor<[5,10],f32> {
+// CHECK-NOT:       tosa.matmul
+// CHECK:           %[[ZERO:.*]] = "tosa.const"() <{values = dense<0.000000e+00> : tensor<5x10xf32>}> : () -> tensor<5x10xf32>
+// CHECK:           %[[RES:.*]] = torch_c.from_builtin_tensor %[[ZERO]] : tensor<5x10xf32> -> !torch.vtensor<[5,10],f32>
+// CHECK:           return %[[RES]]
+func.func @torch.aten.matmul$zero_k_f32(%arg0: !torch.vtensor<[5,0],f32>, %arg1: !torch.vtensor<[0,10],f32>) -> !torch.vtensor<[5,10],f32> {
```
Comment (Lallapallooza) [inline test/Conversion/TorchToTosa/basic.mlir:4901]: Can we add `aten.bmm` coverage as well.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 3]
```diff
@@ -1816,6 +1854,19 @@ class ConvertAtenMatmulBaseOp : public TorchToTosaOpConversionPattern<AtenOpT> {
       return rewriter.notifyMatchFailure(op,
                                          "Matmul: input datatypes mismatched");
 
+    auto resultTy = dyn_cast<RankedTensorType>(
+        OpConversionPattern<AtenOpT>::getTypeConverter()->convertType(
+            op.getType()));
+    if (resultTy && !mlir::tosa::typeHasZeroDim(resultTy) &&
```
Comment (Lallapallooza) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp]: The zero-K shortcut can crash on dynamic result shapes.

A real exported module can produce `aten.mm` with `[M,0] x [0,N] -> [M,N]`; before this PR, the generic zero-dim guard rejected that as unsupported, but now it reaches `getZerosLikeTensor`, which asserts because the result is dynamic. 

Can we gate this path on `resultTy.hasStaticShape()` or add dynamic zero-fill lowering, plus a regression?

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 4]
```diff
@@ -1816,6 +1854,19 @@ class ConvertAtenMatmulBaseOp : public TorchToTosaOpConversionPattern<AtenOpT> {
...
+        hasStaticZeroContraction(lhsTy, rhsTy)) {
+      auto zeroOutput = tosa::getZerosLikeTensor(rewriter, op, resultTy);
```
Comment (Lallapallooza) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:1865]: This fires for quantized path as well? If yes can we add an e2e coverage as well.

**Review by Lallapallooza — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 4 above)
Comment (catcor01) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:1865]: I added int8 zero-K mm e2e coverage, which exercises the integer/TOSA matmul path and this shortcut. I tried the fake-quant qint8-style e2e too, but that fails before matmul on zero-dim dequantize operands.

**Review by Lallapallooza — APPROVED** [review]
LGTM
